modules/git/commit.go

```diff
@@ -466,15 +466,15 @@ func (c *Commit) GetSubModule(entryname string) (*SubModule, error) {
 	return nil, nil
 }
 
-// GetBranchName gets the closes branch name (as returned by 'git name-rev')
+// GetBranchName gets the closes branch name (as returned by 'git name-rev --name-only')
 func (c *Commit) GetBranchName() (string, error) {
-	data, err := NewCommand("name-rev", c.ID.String()).RunInDirBytes(c.repo.Path)
+	data, err := NewCommand("name-rev", "--name-only", c.ID.String()).RunInDirBytes(c.repo.Path)
 	if err != nil {
 		return "", err
 	}
 
-	// name-rev commitID output will be "COMMIT_ID master" or "COMMIT_ID master~12"
-	return strings.Split(strings.Split(string(data), " ")[1], "~")[0], nil
+	// name-rev commitID output will be "master" or "master~12"
+	return strings.Split(strings.Split(string(data), "~")[0], "\n")[0], nil
```

Comment:
Good call on the `--name-only` - I've checked that it's present all the way back to 1.7.2.

I still think the explicit IndexOf is clearer - but I won't force the issue. Just drop the second split as it's quite ugly and set a fixed SplitN.

```suggestion
	return strings.SplitN(strings.TrimSpace(string(data)), "~", 2)[0], nil
```